src/scripts/main.js

```diff
@@ -18,10 +18,40 @@ const food = {
   },
 };
 
+const treeDiv = document.createElement('div');
+
+treeDiv.id = 'tree';
+document.body.appendChild(treeDiv);
+
 const tree = document.querySelector('#tree');
 
 function createTree(element, data) {
-  // WRITE YOUR CODE HERE
+  if (element === null) {
+    throw new Error(`First argument must not be null or undefined`);
+  }
+
+  if (data === undefined) {
+    throw new Error(`Second argument must not be null or undefined`);
+  }
+
+  if (Object.entries(data).length === 0) {
+    return;
+  }
+
+  const ul = document.createElement('ul');
+
+  for (const [key, value] of Object.entries(data)) {
+    const li = document.createElement('li');
+
+    li.textContent = key;
+
+    if (Object.entries(value).length > 0) {
+      createTree(li, value);
+    }
+    ul.appendChild(li);
+  }
```

Comment:
You're creating a new 'ul' element and populating it with 'li' elements for every property in the data object. However, you're not checking if the property value is an object before doing this. This could lead to unexpected behavior if the data object properties are not all objects. You should add a check to ensure that the property value is an object before attempting to create 'li' elements for its properties.